RacingCar/StringCaculator.swift

```diff
@@ -8,15 +8,126 @@
 import Foundation
 
 
-class StringCaculator {
+final class StringCaculator {
     
-    func splitString(to string: String, separator: Character) -> [String] {
-        return string
-                .components(separatedBy: ["(", ")"])
-                .joined()
-                .split(separator: separator).map { String($0)}
+     enum StringCaculatorErrorType: Error {
+        case inputEmptyError
+        case divideByZero
+        case notComputationOperator
+        case notComputationOperand
+        case incorrectExpression
+    }
+    
+    private enum OperatorType: String {
+        case plus = "+"
+        case minus = "-"
+        case multiply = "*"
+        case divide = "/"
+    }
+    
+    func caculate(expression: String) throws -> Int {
+        let removedSpaceExpression: [String] = removeSpaceCharacter(input: expression) // " "제거
+        var validationString = try validateEmptyCheckFor(expression: removedSpaceExpression) // nil값 체크
+        try checkIncorrectExpression(input: expression) // 올바른 표현식인지 체크
+        
+        //처음 값을 current에 저장
+        var current: Int = 0
+        current = try getOperand(operand: validationString.removeFirst())
+        
+        // 순회하여 각 lhs(current) - operator - rhs 계산하여 current에 계속해서 저장
+        while(!validationString.isEmpty) {
+            let lhs: Int = current
+            let `operator`: OperatorType = try getOperator(operator: validationString.removeFirst())
+            let rhs: Int = try getOperand(operand: validationString.removeFirst())
+            current = try caculate(lhs: lhs, operator: `operator`, rhs: rhs)
+        }
+        return current
+    }
+    
+    // 입력값 nil 체크
+    private func validateEmptyCheckFor(expression: [String])  throws -> [String] {
```

Comment:
```suggestion
    private func validateEmptyCheck(with expression: [String])  throws -> [String] {
```

이런식으로 사용할 수 있습니다:)
메서드명에 For, With와 같은 네이밍은 지양해주세요

Reply:
넵 전체적으로 전치사를 사용하지 않도록 하겠습니다!!
전치사가 아예 없는것도 괜찮은 것 같아 없애는 방향으로 하겠습니다!